### PR TITLE
Make optional types optional

### PR DESCRIPTION
**Pull request by saschagrunert**
This adds a few missing types and synchronizes them with the
implementation in containrs. Optional types are now not required any
more which means that all necessary code paths in youki needs to be
adapted as well.

### CODE REVIEW

**Comment by saschagrunert** [comment]
Ooof, sorry for the huge PR.

File: src/cgroups/v1/blkio.rs [hunk 1]
```diff
@@ -106,28 +106,44 @@ impl StatsProvider for Blkio {
 
 impl Blkio {
     fn apply(root_path: &Path, blkio: &LinuxBlockIo) -> Result<()> {
-        for trbd in &blkio.blkio_throttle_read_bps_device {
+        for trbd in blkio
+            .throttle_read_bps_device
+            .as_ref()
+            .context("no throttle_read_bps_device in linux blkio")?
```
Comment (utam0k) [inline src/cgroups/v1/blkio.rs:112]: This is something that should be resolved separately from this PR, but I think there needs to be an implementation for when the config is missing.
I will compile a list of missing implementations of this kind and put them in an issue at a later date.

Comment (utam0k) [reply src/cgroups/v1/blkio.rs:112]: https://github.com/containers/youki/issues/184

**Review by utam0k — CHANGES_REQUESTED** [review]
Can I ask you to check the default test in the runtime-tools integration test, because it seems to take infinite time and the test is not progressing?
I can check it tomorrow night on my end.

**Comment by saschagrunert** [comment]
Yeah the issue is now that with making the types optional, we also have to make the code paths in youki optional. Right now the integrations tests will fail because for example:
```
[DEBUG src/container/builder_impl.rs:88] 2021-08-05T12:11:04.943893124+02:00 failed to run container_init: Failed to prepare rootfs

Caused by:
    no mount_label in spec
```

**Comment by utam0k** [comment]
@saschagrunert Thanks for your report. I have started to fix it based on your report, but I need more time, so please wait a few days.
https://github.com/saschagrunert/youki/pull/1

**Comment by saschagrunert** [comment]
> @saschagrunert Thanks for your report. I have started to fix it based on your report, but I need more time, so please wait a few days.
> [saschagrunert#1](https://github.com/saschagrunert/youki/pull/1)

Oh wow, cool thank you! I was wondering if making it optional is the right approach, but when it comes to serializing then I still think that having them as `Option` types is the correct way to represent the data.

**Comment by utam0k** [comment]
@saschagrunert 
Sorry, I missed a few things and it wasn't working. I sent out a PR to fix it.
https://github.com/saschagrunert/youki/pull/2

**Comment by utam0k** [comment]
Hmm, I'll try exactly that in my forked repository later.

**Comment by saschagrunert** [comment]
Gave it another rebase because of the conflicts

**Comment by utam0k** [comment]
@saschagrunert 
I'm sorry for repeating myself. I'm sure this time!
https://github.com/saschagrunert/youki/pull/3

**Comment by utam0k** [comment]
@saschagrunert 
I think my patch has been merged, but it is not reflected in this PR. Hmmm... do you have any idea what might be causing this?

**Comment by saschagrunert** [comment]
> @saschagrunert
> I think my patch has been merged, but it is not reflected in this PR. Hmmm... do you have any idea what might be causing this?

GitHub had some issues yesterday, I'll re push the merge commit.

**Review by utam0k — APPROVED** [review]
Thasnk!

**Comment by saschagrunert** [comment]
I think we can now transfer the source code into https://github.com/containers/oci-spec-rs, do you wanna take care of that?

**Comment by utam0k** [comment]
> I think we can now transfer the source code into [containers/oci-spec-rs](https://github.com/containers/oci-spec-rs), do you wanna take care of that?

Yes, I will try it around this weekend.

**Pull request merged**